### PR TITLE
Percolation patch

### PR DESCRIPTION
**Pull request by ew3361zh**
This PR seeks to address issue #805 by:

1. Rearranging the logic in the `Percolation.percolate()` method so that the the percolated water is added to the layer below the day after it's percolated from the layer above rather than the same day (comments added for clarity in comment but removed in actual code):
Screenshot 2023-10-02 at 11 10 19 AM
2. Updating unit testing for the `percolate()` method.

Notes:
There was previously only integration testing for the `percolate()` method and while @Isgotkowitz and I tried to dissect the logic, we couldn't quite figure it out. I wrote a new unit test more specifically for the `percolate()` method and tested whether it achieved the same level of coverage for percolation.py if the integration test was removed and coverage held steady at 100%. I don't know what other .py files the `test_percolate()` integration test may affect though so I am leaving it in place for the time being. If reviewers would prefer it be removed because it's not certain what it's testing, please comment that.

### CODE REVIEW

File: RUFAS/routines/field/soil/percolation.py [hunk 1]
```diff
@@ -58,7 +58,9 @@ def percolate(self, has_seasonal_high_water_table: bool) -> None:
                 percolated_water = self._percolate_between_layers(self.data.time_step, current_layer, layer_below)
                 current_layer.water_content -= percolated_water
                 current_layer.percolated_water = percolated_water
-                layer_below.water_content += percolated_water
+                if layer_number > 0:
+                    layer_above = self.data.soil_layers[layer_number - 1]
+                    current_layer.water_content += layer_above.percolated_water
```
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/percolation.py:63]: This works for all the soil layers contained in `self.data.soil_layers`, but water doesn't percolate into the vadose zone layer.

**Review by Isgotkowitz — CHANGES_REQUESTED** [review]
This is a good start but the vadose zone layer needs to be percolated into.

**Comment by Isgotkowitz** [comment]
> Notes:
There was previously only integration testing for the percolate() method and while @Isgotkowitz and I tried to dissect the logic, we couldn't quite figure it out. I wrote a new unit test more specifically for the percolate() method and tested whether it achieved the same level of coverage for percolation.py if the integration test was removed and coverage held steady at 100%. I don't know what other .py files the test_percolate() integration test may affect though so I am leaving it in place for the time being. If reviewers would prefer it be removed because it's not certain what it's testing, please comment that.

@ew3361zh thank you for writing this test. I am working on another bug in percolation right now, and when I finalize the PR for it I will update `test_percolate()` to work in a correct and understandable manner.

File: RUFAS/routines/field/soil/percolation.py (hunk 1 above)
Comment (Isgotkowitz) [reply RUFAS/routines/field/soil/percolation.py:63]: I think just adding `self.data.vadose_zone_layer.water_content += self.data.soil_layers[-1].percolated_water` after the for loop has finished executing would work.

Comment (ew3361zh) [reply RUFAS/routines/field/soil/percolation.py:63]: I tried it with the line after the loop but the testing wasn't behaving as expected so added the line within the loop based on some conditional logic which I think does very close to what it does outside the loop but behaves more as expected with water_content and percolated_water amounts by layer.

**Review by Isgotkowitz — APPROVED** [review]
LGTM. Thank you Niko

**Review by matthew7838 — APPROVED** [review]
Niko, nice change and good coverage on the tests.

**Pull request merged**